planning/autoware_path_optimizer/src/node.cpp

```diff
@@ -391,8 +391,8 @@ std::vector<TrajectoryPoint> PathOptimizer::optimizeTrajectory(const PlannerData
   const bool elapsed_time_over_three_seconds = (elapsed_time > 3.0);
 
   auto optimized_traj_points =
-    optimized_traj_failed && elapsed_time_over_three_seconds ? p.traj_points : std::move(*mpt_traj);
-  is_optimization_failed_ = optimized_traj_failed && elapsed_time_over_three_seconds;
+    optimized_traj_failed || elapsed_time_over_three_seconds ? p.traj_points : std::move(*mpt_traj);
```

Comment:
For this fix, is 'elapsed_time_over_three_seconds' even needed?

Reply:
```suggestion
     auto optimized_traj_points = [&]() {
    if (optimized_traj_failed && elapsed_time_over_three_seconds) {
      return p.traj_points;
    } else if (optimized_traj_failed && !elapsed_time_over_three_seconds) {
      return getPrevOptimizedTrajectory(p.traj_points);
    } else {
      return std::move(*mpt_traj);
    }
  }();
```

I think this might be better. The idea is to only return mpt_traj if the optimization is successful.

If the optimization is unsuccessful, but 3s haven't elapsed: return the previous MPT result
If the optimization is unsuccessful, but 3s have elapsed: return the smoother output (p.traj_points)